db/migrations/20240318200010-update-exisiting-remediation-names.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+module.exports = {
+    up: async (q, {STRING}) => {
+        const transaction = await q.sequelize.transaction()
+        try {
+            //===================================================
+            // This query will update existing remediation names to ensure that there are
+            // no duplicate remediation names within an organization
+            //===================================================
+
+            // Find duplicate remediation names within an organization and add the username to the beginning of the name
+            // Example: '2024-01-02T06:06:55.759Z: duplicate name', '2024-02-19T16:23:47.267Z: duplicate name', '2023-11-27T16:29:40.969Z: duplicate name'...
+            const update_dups_in_org = `UPDATE remediations
+                                        SET name = created_at || ': ' || name 
+                                        WHERE (name, tenant_org_id) IN
+                                        (
+                                            SELECT name, tenant_org_id
```

Comment:
I think:
```
SELECT
  coalesce(name, 'Unnamed Remediation Plan') AS name,
  tenant_ord_id
...
```
would also include the entries where `name` is NULL if you do something like:
```
HAVING COUNT(name) > 1 OR name IS NULL
```

Reply:
I think this would work too or even this:
```
UPDATE remediations
SET name = created_at || ': ' || name 
WHERE (name, tenant_org_id) IN
(
           SELECT name, tenant_org_id
           FROM remediations
           GROUP BY name, tenant_org_id
           HAVING ( COUNT( name ) > 1 )
)
OR name IS NULL
```

Reply:
Here's my latest absurd suggestion:
```
WITH dups AS (
    SELECT
        a.id,
        a.name,
        a.tenant_org_id,
        ROW_NUMBER() OVER (PARTITION BY a.tenant_org_id, a.name) AS number
    FROM remediations a
    JOIN (
        SELECT name, tenant_org_id
        FROM remediations
        GROUP BY name, tenant_org_id
        HAVING COUNT(*) > 1 OR name IS NULL
        LIMIT 1000
    ) b
    ON (a.name = b.name OR (a.name IS NULL AND b.name IS NULL)) AND a.tenant_org_id = b.tenant_org_id
)
UPDATE remediations
    SET name = coalesce(dups.name, 'Unnamed Remediation Plan') || ' - ' || dups.number
FROM dups
WHERE dups.id = remediations.id;
```
We could omit the `LIMIT` for now and see if it works in one pass.  If not, we can add it and then just repeat this until no more rows are changed.

This also goes back to using incrementing numbers instead of awkward timestamps.